src/anomalib/models/video/fuvas/torch_model.py

```diff
@@ -0,0 +1,218 @@
+        if backbone in {"i3d_r50", "x3d_l", "x3d_xs", "x3d_s", "x3d_m"}:
+            # Load PyTorchVideo model from torch hub
+            # Supported models: i3d_r50, x3d_l, x3d_xs, x3d_s, x3d_m
+            # Common input sizes: (448,512)
+            # Common feature extraction layers:
+            #   - blocks.6.dropout: 2048 dimensional features
+            #   - blocks.5.res_blocks.2: Features before final layers
+            net = torch.hub.load("facebookresearch/pytorchvideo", model=backbone, pretrained=pre_trained)
+
+        elif backbone == "swin3d_b":
+            # Load Swin3D-B model pre-trained on Kinetics-400 and ImageNet-22K
+            # Alternative weights: KINETICS400_V1 (pre-trained on Kinetics-400 only)
+            # Input normalization:
+            #   - KINETICS400_IMAGENET22K_V1: Uses ImageNet normalization
+            #   - KINETICS400_V1: mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]
+            # Common input size: (256,256), minimum temporal dimension: 1
+            net = swin3d_b(weights=Swin3D_B_Weights.KINETICS400_IMAGENET22K_V1)
+        else:
+            msg = (
+                f"Backbone '{backbone}' is not supported. "
+                f"Use one of: 'i3d_r50', 'x3d_l', 'x3d_xs', 'x3d_s', 'x3d_m', 'swin3d_b'."
+            )
+            raise ValueError(msg)
+
+        net.eval()
+        self.feature_extractor = create_feature_extractor(net, return_nodes=[layer])
+
+        self.peak = 0
+
+    def fit(self, dataset: torch.Tensor) -> None:
+        """Fit PCA model to dataset.
+
+        Args:
+            dataset (torch.Tensor): Input dataset with shape
+                ``(n_samples, n_features)``.
+        """
+        self.pca_model.fit(dataset.numpy())
+        oi_j = self.pca_model.transform(dataset.numpy())
+        oi_recon = self.pca_model.inverse_transform(oi_j)
+        self.peak = abs(dataset.numpy() - oi_recon)
+
+    def score(self, features: torch.Tensor, feature_shapes: tuple) -> torch.Tensor:
+        """Compute anomaly scores.
+
+        Scores are PCA-based feature reconstruction error (FRE) scores.
+
+        Args:
+            features (torch.Tensor): Features for scoring with shape
+                ``(n_samples, n_features)``.
+            feature_shapes (tuple): Shape of features tensor for anomaly map.
+
+        Returns:
+            tuple[torch.Tensor, Optional[torch.Tensor]]: Tuple containing
+                (scores, anomaly_maps).
+        """
+        feats_projected = self.pca_model.transform(features.numpy())
+        feats_reconstructed = self.pca_model.inverse_transform(feats_projected)
+        fre_prereshape = np.square(features.numpy() - feats_reconstructed)
+        fre_prereshape_np = fre_prereshape
+        fre = fre_prereshape_np.reshape(feature_shapes)
+        score_map_np = np.sum(fre, axis=(1, 2))  # NxTxCxHxW->NxHxW
+        score_map = torch.from_numpy(score_map_np)
+        score = torch.from_numpy(np.sum(score_map_np, axis=(1, 2)))  # NxHxW->N
```

Comment:
Would it be possible to write this all in torch without converting to numpy back and forth?
This would reduce the dependency on numpy, also could speed it up a bit